hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/metrics/Metrics.java

```diff
@@ -0,0 +1,106 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.core.metrics;
+
+import com.hazelcast.jet.Job;
+import com.hazelcast.jet.impl.metrics.MetricsImpl;
+
+/**
+ * Utility class for obtaining handlers to user-defined metrics.
+ * <p>
+ * User-defined metric are simple numeric values used to count or measure
+ * things. A code submitted to Jet can use them to publish any custom run-time
+ * values.
+ * <p>
+ * This class provides the means for creating of handlers
+ * for setting up and manipulating the user-defined metrics.
+ */
+@SuppressWarnings("WeakerAccess")
+public final class Metrics {
+
+    private Metrics() {
+    }
+
+    /**
+     * Returns a non-thread-safe handler for manipulating the metric with the
+     * specified name.
+     * <p>
+     * This method only works if called from a processor thread and the
+     * returned handler is tied to the processor that is currently executing.
+     * The handler is created on the first call, subsequent calls return the
+     * cached instance and ignore the requested unit or thread safety. Until
+     * the first call is made, the metric is not visible in JMX or using {@link
+     * Job#getMetrics()}.
+     */
+    public static Metric metric(String name) {
+        return MetricsImpl.metric(name, Unit.COUNT);
+    }
+
+    /**
+     * Returns a non-thread-safe handler for manipulating the metric with the
+     * specified name and measurement unit.
+     * <p>
+     * This method only works if called from a processor thread and the
+     * returned handler is tied to the processor that is currently executing.
+     * The handler is created on the first call, subsequent calls return the
+     * cached instance and ignore the requested unit or thread safety. Until
+     * the first call is made, the metric is not visible in JMX or using {@link
+     * Job#getMetrics()}.
+     */
+    public static Metric metric(String name, Unit unit) {
+        return MetricsImpl.metric(name, unit);
+    }
+
+    /**
+     * Returns a thread-safe handler for manipulating the metric with the
+     * specified name and measurement unit.
+     * <p>
+     * You need the thread-safe method if you submit work to other threads and
+     * want to manipulate the metric in those threads. For example:
+     * <pre>
+     * {@code
+     *  p.drawFrom(...)
+     *   .filterUsingServiceAsync(
+     *       ServiceFactory.withCreateFn(i -> "foo"),
+     *       (ctx, item) -> {
+     *           Metric dropped = Metrics.threadSafeMetric("dropCount", Unit.COUNT);
+     *           return CompletableFuture.supplyAsync(
+     *               () -> {
+     *                   // the callback runs on another thread
+     *                   boolean pass = item % 2L == 0;
+     *                   if (!pass) {
+     *                       dropped.inc();
```

Comment:
Old name

Reply:
I will call you Viliam Eagle-Eye from now on!